### PR TITLE
Provide a '--help' option to all our scripts

### PR DESCRIPTION
**Pull request by okurz**
Mojo services already have an internal help options but in some cases we
could not reach it so we should treat '--help' in a special way to allow
users to reach help :)

This also removes all TODOs from t/44-scripts.t

### CODE REVIEW

**Comment by kalikiana** [comment]
I presume this fixes [#65933](https://progress.opensuse.org/issues/65933)

File: lib/OpenQA/LiveHandler.pm [hunk 1]
```diff
@@ -47,6 +47,7 @@ sub startup {
     my $self = shift;
 
     $self->defaults(appname => 'openQA Live Handler');
+    return if grep { /--help/ } @ARGV;
```
Comment (kalikiana) [inline lib/OpenQA/LiveHandler.pm:50]: Is this sensible to do within [startup](https://mojolicious.org/perldoc/Mojolicious#startup)? The docs don't really say anything about that. What about exiting with an error code?

Comment (perlpunk) [reply lib/OpenQA/LiveHandler.pm:50]: I'm not completely happy with this change, too. For example help can also be called with `-h`.

Comment (okurz) [reply lib/OpenQA/LiveHandler.pm:50]: If I would have found a better approach I would have done it :) I am happy to change if you have good suggestions. I could of course change the regex to `/^(-h|--help)$/`

Comment (kalikiana) [reply lib/OpenQA/LiveHandler.pm:50]: I suspect @kraih would be best to comment on that then?

Comment (okurz) [reply lib/OpenQA/LiveHandler.pm:50]: Certainly. But I am hesitant to ask individuals for review to not annoy :)

Comment (kraih) [reply lib/OpenQA/LiveHandler.pm:50]: No, this is not a good idea at all. Same reason you shouldn't add defaults to the app scripts.

Comment (kraih) [reply lib/OpenQA/LiveHandler.pm:50]: And there are more ways to do `--help`, like `./script/foo help daemon` and the like.

**Comment by kraih** [comment]
Don't mess with the Mojolicious command system. This will have unintended side effects.

**Comment by okurz** [comment]
Do you have any better suggestion then? Or should we just delete the "TODO" entries from test and skip the mojo commands and let them fail as they currently do unless called with correct user?

**Comment by kraih** [comment]
The real question here is why the default `--help` doesn't work anymore. All Mojolicious apps actually start out having that feature. Maybe some code needs to be moved into a `before_server_start` hook? Do you know what exactly breaks `--help` currently?

**Comment by kraih** [comment]
I tried the scripts locally, and `openqa-livehandler --help` works fine. Just `openqa-workercache --help` fails with an exception because of a missing config option (that could probably be handled better).

**Comment by perlpunk** [comment]
> I tried the scripts locally, and `openqa-livehandler --help` works fine.

Yeah, but not in CI:
https://github.com/os-autoinst/openQA/pull/2967
```
#   Failed test 'Calling 'openqa-livehandler --help' returns exit code 0'
#   at t/44-scripts.t line 42.
#          got: '512'
#     expected: '0'
# Output: Can not create MOJO_TMPDIR : mkdir /var/lib/openqa: Permission denied at /home/squamata/project/script/../lib/OpenQA/Setup.pm line 243.
# 
# Can't open database lock file /var/lib/openqa/db/db.lock! at /home/squamata/project/script/../lib/OpenQA/Schema.pm line 87.
```

**Comment by okurz** [comment]
Yes, dependencies are part of the problem which I am circumventing

**Comment by kraih** [comment]
Ah, so it's directory initialization for both... not sure what the best solution for that is. It's definitely not hijacking the command system though. Might even be worth thinking about an upstream change in Mojolicious.

**Comment by kraih** [comment]
This is a really complicated problem as it turns out. Without upstream changes there are only two solutions for now (at least that i can think of). 1) we can defer those initialization operations somehow, perhaps with a lazy attribute on the app class, or 2) a hack like you proposed, although `return if $ENV{MOJO_HELP};` would work better (private use ENV vars might disappear in the future though)...

**Comment by okurz** [comment]
I never argued my solution would be the "best" ;) So what do you recommend to get rid of the "TODOs" in the ticket? Should we just delete the "TODO" reference when excluding the scripts in the test? Or should we not look for "--help" but ensure that the scripts are called with the right user?

**Comment by kraih** [comment]
Just accepting that these apps need to be properly configured before you can run them (even with `--help` flag) might be the most sane choice. Otherwise if we really really need `--help` to work, the env var hack with a huge comment explaining why would probably be best for now.

**Comment by okurz** [comment]
Done the "comment + env" path. I think this is good enough now and the easy way out :) I think checking the permissions and correct users is tricky because this very much depends on the environment. It is rather a packaging specific approach that triggers the common error of "permission denied" for us, not necessarily something that is prescribed by the perl implementation.

Actually originally I was just calling the stuff and trying to introspect what attributes I could read and found nothing better than @ARGV because I could not see how the modules communicate to Mojo base that help is requested. Haven't thought about the interesting choice of an environment variable ;)

**Pull request merged**